api/types/saml_idp_service_provider.go

```diff
@@ -161,6 +185,26 @@ func (s *SAMLIdPServiceProviderV1) CheckAndSetDefaults() error {
 		s.Spec.EntityID = ed.EntityID
 	}
 
+	if len(s.GetAttributeMapping()) > 0 {
+		attrNames := make([]string, 0)
```

Comment:
Nit: Would probably be a little bit more efficient to use a map here instead of a slice.